staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -140,6 +141,16 @@ func (config *DirectClientConfig) ClientConfig() (*restclient.Config, error) {
 		clientConfig.Timeout = timeout
 	}
 
+	if configClusterInfo.ConnectionTimeout != "" {
+		connectionTimeout, err := ParseTimeout(configClusterInfo.ConnectionTimeout)
+		if err != nil {
+			return nil, err
+		}
+		clientConfig.ConnectionTimeout = connectionTimeout
+	} else {
+		clientConfig.ConnectionTimeout = 7 * time.Second
```

Comment:
@smarterclayton Thanks for the feedback. I've now made this 7-second default apply only to the client that the Kubelet uses. All other users of `client-go` now do not have this functionality by default.